src/atomEffect.ts

```diff
@@ -21,21 +21,26 @@ export function atomEffect(
     refreshAtom.debugPrivate = true
   }
 
-  const initAtom = atom(null, (get, set, mounted: boolean) => {
-    const ref = get(refAtom)
-    if (mounted) {
-      ref.mounted = true
-      set(refreshAtom, (c) => c + 1)
-    } else {
-      ref.cleanup?.()
-      ref.cleanup = undefined
-      ref.mounted = false
+  const initAtom = atom(
+    (get, { setSelf }) => {
+      const ref = get(refAtom)
+      if (ref.mounted) return
+      Promise.resolve().then(() => {
+        setSelf(true)
+      })
```

Comment:
I don't think this violates the usage, but not sure if it causes another pitfall.

Reply:
so ref atom will not change value so initAtom read should not recompute. Is this guaranteed? 
Anyways, is shouldn't be an issue to run multiple times.